lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -320,6 +321,122 @@ sub worker_id {
     return 0;
 }
 
+sub reschedule_state {
+    my $self = shift;
+    my $state = shift // OpenQA::Schema::Result::Jobs::SCHEDULED;
+
+    # cleanup
+    $self->set_property('JOBTOKEN');
+    $self->release_networks();
+    $self->owned_locks->delete;
+    $self->locked_locks->update({locked_by => undef});
+
+    $self->update(
+        {
+            state              => $state,
+            t_started          => undef,
+            assigned_worker_id => undef,
+            result             => NONE
+        });
+    if ($self->worker) {
+        # free the worker
+        $self->worker->update({job_id => undef});
+    }
+    if ($self->state eq $state) {
+        #if (!$self->assigned_worker_id && !$self->worker) {
```

Comment:
this PR is meant so you can check more the logic behind here, and detect bugs in that. e.g. it will make a stronger bond to check if the worker state is assigned here? if we don't care then i drop the comment -  the cleanup/documentation cames after ( since it was a WIP :) )